### PR TITLE
docs/executors: Add more troubleshooting items, and cleanup some docs

### LINKED ISSUE #46216: Document Troubleshooting Guide

### PR DESCRIPTION
**Pull request by sanderginn**
- Closes #46216

Adds several troubleshooting items, and cleans up formatting issues in a few pages.

## Test plan
Validated with `sg run docsite`

### CODE REVIEW

File: doc/admin/executors/executors_troubleshooting.md [hunk 1]
```diff
@@ -47,3 +62,120 @@ If a server-side batch change fails unexpectedly, it's possible to recreate the
...
+Run the following command on the executor instance to determine whether it responds properly:
+```shell
+# If EXECUTOR_DOCKER_REGISTRY_MIRROR_URL is set to a custom URL, replace the base endpoint with its value
+curl http://localhost:5000/v2/_catalog
+```
+    
+### Registry is mounted in the file system
+Verify that the registry is mounted under the expected path in the file system by running:
+```shell
+ls /mnt/registry/docker/registry/v2/repositories/sourcegraph
```
Comment (Piszmog) [inline doc/admin/executors/executors_troubleshooting.md:101]: Will this path exist even if the executor has not processed any Jobs yet? (it has just been running idle for a while)

File: doc/admin/executors/executors_troubleshooting.md [hunk 2]
```diff
@@ -47,3 +62,120 @@ If a server-side batch change fails unexpectedly, it's possible to recreate the
...
+curl http://localhost:5000/v2/_catalog
+```
+    
+### Registry is mounted in the file system
+Verify that the registry is mounted under the expected path in the file system by running:
+```shell
+ls /mnt/registry/docker/registry/v2/repositories/sourcegraph 
+```
+
+## Connecting to cloud provider executor instances
```
Comment (Piszmog) [inline doc/admin/executors/executors_troubleshooting.md:104]: Might be helpful to also give instructions on how to connect to the instances via the cloud provider UIs (if possible).

**Review by Piszmog — APPROVED** [review]
Looks good! Thank you! Just some minor comments/questions

**Comment by sanderginn** [comment]
Automerged, but I'll address the feedback in a separate PR

File: doc/admin/executors/executors_troubleshooting.md [hunk 3]
```diff
@@ -47,3 +62,120 @@ If a server-side batch change fails unexpectedly, it's possible to recreate the
 1. Run `ignite ps` to list all currently running VMs
 1. Run `ignite attach <vm id>` to get a shell to the running VM
 
+## List preferred Linux distros
+An ARM64 (x86_64) linux distro must be used due to the machine type of the VM. You may list available ARM64 distros with the following command, depending on your cloud provider:
+
+### GCP    
+```shell
+gcloud compute images list --filter='(family~amd)'
```
[the remark this answers is not in the document]
Comment (sanderginn) [reply doc/admin/executors/executors_troubleshooting.md:70]: I'm operating on the assumption that the prerequisite for Firecracker support is an ARM64 machine, and this command returns available images for that architecture. So I would say that these images therefore support Firecracker. @eseliger that makes sense?

**Pull request merged**